examples/cloud-datafusion-functions-plugins/README.md

```diff
@@ -0,0 +1,374 @@
+- [Reusable Plugins](#reusable-plugins-for-cloud-data-fusion-cdf--cdap)
+    - [Overview](#overview)
+    - [CheckPointReadAction, CheckPointUpdateAction](#checkpointreadaction-checkpointupdateaction)
+        - [Dependencies](#dependencies)
+            - [Setting up Firestore](#setting-up-firestore)
+            - [Set Runtime Arguments](#set-runtime-arguments)
+    - [CopyTableAction](#copytableaction)
+    - [DropTableAction](#droptableaction)
+    - [TruncateTableAction](#truncatetableaction)
+- [Putting it all together into a Pipeline](#putting-it-all-together-into-a-pipeline)
+  - [CheckPointReadAction](#checkpointreadaction)
+  - [TruncateTableAction](#truncatetableaction)
+  - [Database source](#database-source)
+  - [BigQuery sink](#bigquery-sink)
+  - [MergeLastUpdateTSAction](#mergelastupdatetsaction)
+  - [CheckPointUpdateAction](#checkpointupdateaction)
+- [Building the CDF/CDAP Plugin (JAR file / JSON file) and deploying into CDF/CDAP](#building-the-cdfcdap-plugin-jar-file--json-file-and-deploying-into-cdfcdap)
+
+# Reusable Plugins for Cloud Data Fusion (CDF) / CDAP
+
+## Overview
+The CDF/CDAP plugins detailed below can be reused in the context of data pipelines.
+
+Let's say you run your incremental pipeline once every 5 minutes. When running an incremental pipeline, you have to filter the records by a specific field (e.g., `lastUpdateDateTime` of records > latest watermark value - buffer time) so it will sync the records that were updated since your last incremental sync. Subsequently, a merge and dedupe step is done to make sure only new/updated are synced into the destination table.
+
+## CheckPointReadAction, CheckPointUpdateAction
+
+**Plugin Description**  
+Creates, reads, and updates checkpoints in incremental pull pipelines.
+
+
+`CheckPointReadAction` - reads checkpoint in Firestore DB and provides the data during runtime as environment variable
+
+`CheckPointUpdateAction` - updates CheckPoint in Firestore DB (i.e., creates a new document and stores maximum update date / time  from BQ so the next run it can use this checkpoint value to filter records that were added since then)
```

Comment:
What is the correct spelling? "Checkpoints" or "Checkpoint"? You use these names inconsistently. See paragraph earlier.

Reply:
Well, it depends - it just the "Checkpoints" is just the plural form of "Checkpoint". 
That said, I do see your point - I'll try to make it consistent. Thank you.

Reply:
Sorry. I mean "Checkpoint" and "CheckPoint".

Reply:
> reads checkpoint

> updates CheckPoint

Reply:
got it - thanks for clarifying